src/pages/insights/azure-storage-firewalls.mdx

```diff
@@ -0,0 +1,97 @@
+
+#### Warning!
+
+There is a strange quirk in Azure storage that is highlighted in the [Restrictions for IP network rules](https://learn.microsoft.com/en-us/azure/storage/common/storage-network-security?tabs=azure-portal#restrictions-for-ip-network-rules) section of the documentation.
+
+> You can't use IP network rules in the following cases:
+>
+> * To restrict access to clients in same Azure region as the storage account. IP network rules have no effect on requests that originate from the same Azure region as the storage account. Use Virtual network rules to allow same-region requests.
+> * To restrict access to clients in a paired region that are in a virtual network that has a service endpoint.
+> * To restrict access to Azure services deployed in the same region as the storage account. Services deployed in the same region as the storage account use private Azure IP addresses for communication. So, you can't restrict access to specific Azure services based on their public outbound IP address range.
+
+Traffic inside the same Azure region (ex. a virtual machine accessing the storage account) uses an Azure private IP which is hidden from the users. The same applies to resources in the paired region (if there is one) if the virtual network has service endpoints enabled. This is essentially how service endpoints work behind the scenes. These internal IPs cannot be blocked by IP rules. In order to control this access, virtual network rules and service endpoints must be used. Otherwise the only option is to disable public access completely.
+
+### Resource Instances
+
+Certain resources can be permitted to bypass the firewall. A full list of supported resources are available in the Azure Portal. These can support a specific instance or any instance, including future instances, in the same resource group, subscription, or Entra tenant as chosen.
+
+### Exceptions
+
+One of the three exception checkboxes is for `Azure Trusted Services`. The list of [default trusted services](https://learn.microsoft.com/en-us/azure/storage/common/storage-network-security?tabs=azure-portal#trusted-access-for-resources-registered-in-your-subscription) is relatively short and include Azure Backup for unmanaged disks and Azure Networking logs for example.
+
+However, there is a second list for [trusted access based on managed identity](https://learn.microsoft.com/en-us/azure/storage/common/storage-network-security?tabs=azure-portal#trusted-access-based-on-a-managed-identity) which permits a much wider list of services as long as those services authenticate with Azure AD. This list includes services such as Azure Databricks Access Connectors allowing Databricks to read and write from an internal data lake or Azure Databricks to read and write data from a Microsoft provided integration runtime. This list is a much broader way of using the `Resource Instances` feature when the services may be managed in another tenant such as a Microsoft provided service.
```

Comment:
"Entra" surely, rather than Azure AD. (I know you hate it, but it is what it is)